app/graphql/types/git/directory_entry_type.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/BlockLength
+Types::Git::DirectoryEntryType = GraphQL::InterfaceType.define do
+  # rubocop:enable Metrics/BlockLength
+  name 'DirectoryEntry'
+  description 'A directory entry (directory or file) of a repository'
+
+  field :name, !types.String do
+    description 'The name of the entry'
+  end
+
+  field :path, !types.String do
+    description 'The path of the entry'
+  end
+
+  # rubocop:disable Metrics/BlockLength
+  field :log, !types[!Types::Git::CommitType] do
```

Comment:
What ordering will this have? Newest to oldest or the other way around? My intuition would be newest to oldest, but that would mean it's the other way around than all other paginated lists we have in the API.

I've seen other APIs use `first n` and `last n` with skip, so that you can have both orderings.